app/controllers/concerns/switch_locale.rb

```diff
@@ -16,6 +17,18 @@ def switch_locale_using_account_locale(&)
     set_locale(locale, &)
   end
 
+  def locale_from_portal(&)
+    return if params[:locale]
+
+    domain = request.host
+    return if domain == URI.parse(ENV.fetch('FRONTEND_URL', '')).host
+
+    @portal = Portal.find_by(custom_domain: domain)
+    return unless @portal
+
+    @portal.default_locale
+  end
```

Comment:
@sojan-official need your thoughts on this

Reply:
Should also check for `HELPCENTER_URL` as well. But makes sense.

Reply:
This is specifically for the root url page, there is appropriate validation for the help center route for this